Source/CesiumRuntime/Private/CesiumGeoreference.cpp

```diff
@@ -605,200 +595,151 @@ void ACesiumGeoreference::Tick(float DeltaTime) {
 }
 
-glm::dvec3 ACesiumGeoreference::TransformLongitudeLatitudeHeightToUe(
+glm::dvec3 ACesiumGeoreference::TransformLongitudeLatitudeHeightToUnreal(
     const glm::dvec3& longitudeLatitudeHeight) const {
-  glm::dvec3 ecef =
-      this->TransformLongitudeLatitudeHeightToEcef(longitudeLatitudeHeight);
-  return this->TransformEcefToUe(ecef);
+  return this->_geoTransforms.TransformLongitudeLatitudeHeightToUnreal(
+      CesiumActors::getWorldOrigin4D(this),
+      longitudeLatitudeHeight);
 }
 
-FVector ACesiumGeoreference::InaccurateTransformLongitudeLatitudeHeightToUe(
+FVector ACesiumGeoreference::InaccurateTransformLongitudeLatitudeHeightToUnreal(
     const FVector& longitudeLatitudeHeight) const {
-  glm::dvec3 ue = this->TransformLongitudeLatitudeHeightToUe(glm::dvec3(
-      longitudeLatitudeHeight.X,
-      longitudeLatitudeHeight.Y,
-      longitudeLatitudeHeight.Z));
+  glm::dvec3 ue = this->_geoTransforms.TransformLongitudeLatitudeHeightToUnreal(
+      CesiumActors::getWorldOrigin4D(this),
+      VecMath::createVector3D(longitudeLatitudeHeight));
   return FVector(ue.x, ue.y, ue.z);
 }
 
-glm::dvec3 ACesiumGeoreference::TransformUeToLongitudeLatitudeHeight(
+glm::dvec3 ACesiumGeoreference::TransformUnrealToLongitudeLatitudeHeight(
     const glm::dvec3& ue) const {
-  glm::dvec3 ecef = this->TransformUeToEcef(ue);
-  return this->TransformEcefToLongitudeLatitudeHeight(ecef);
+  return this->_geoTransforms.TransformUnrealToLongitudeLatitudeHeight(
+      CesiumActors::getWorldOrigin4D(this),
+      ue);
 }
 
-FVector ACesiumGeoreference::InaccurateTransformUeToLongitudeLatitudeHeight(
+FVector ACesiumGeoreference::InaccurateTransformUnrealToLongitudeLatitudeHeight(
     const FVector& ue) const {
   glm::dvec3 llh =
-      this->TransformUeToLongitudeLatitudeHeight(glm::dvec3(ue.X, ue.Y, ue.Z));
+      this->_geoTransforms.TransformUnrealToLongitudeLatitudeHeight(
+          CesiumActors::getWorldOrigin4D(this),
+          VecMath::createVector3D(ue));
   return FVector(llh.x, llh.y, llh.z);
 }
 
 glm::dvec3
-ACesiumGeoreference::TransformEcefToUe(const glm::dvec3& ecef) const {
-  glm::dvec3 ueAbs = this->_ecefToUeAbs * glm::dvec4(ecef, 1.0);
-
-  const FIntVector& originLocation = this->GetWorld()->OriginLocation;
-  return ueAbs -
-         glm::dvec3(originLocation.X, originLocation.Y, originLocation.Z);
+ACesiumGeoreference::TransformEcefToUnreal(const glm::dvec3& ecef) const {
+  return this->_geoTransforms.TransformEcefToUnreal(
+      CesiumActors::getWorldOrigin4D(this),
+      ecef);
 }
 
-FVector
-ACesiumGeoreference::InaccurateTransformEcefToUe(const FVector& ecef) const {
-  glm::dvec3 ue = this->TransformEcefToUe(glm::dvec3(ecef.X, ecef.Y, ecef.Z));
+FVector ACesiumGeoreference::InaccurateTransformEcefToUnreal(
+    const FVector& ecef) const {
+  glm::dvec3 ue = this->_geoTransforms.TransformEcefToUnreal(
+      CesiumActors::getWorldOrigin4D(this),
+      glm::dvec3(ecef.X, ecef.Y, ecef.Z));
```

Comment:
```suggestion
      VecMath::createVector3D(ecef);
```

Reply:
There is likely a ton of places where overly elaborate `something[index]`- and `something.x/y/z`-twiddling can/could/should be replaced with the corresponding `VecMath` function. **Iff** we wanted to do this dedicatedly, it should be one, orchestrated review pass of the whole codebase. Otherwise, it may be replaced successively, whenever somebody touches a code part and notices "Hey, that could be replaced...."